wordpress-mobile/WordPress-Android
# Offline Support Pages : Cancel Auto Upload

Fixes #11150 

## Findings
The `Post List` has logic within `PostListItemUiStateHelper` that helps to determine which buttons should be shown based on several different conditions. This logic was adapted to create a similar behavior for the `Page List` cancel auto-upload functionality. 

## Solution
The `PageItems` had the actions hardcoded within its constructor. The actions were removed from them and added at instantiation so that logic could be run that determines if the `Cancel` button should be shown or not. 

## Issues Noticed 
* There's a crash that occurs at times when trying to fetch the post to determine if a page can be canceled or not, so a null check has to be done 0a035ad6de396287865228b0acd0382da4389669. I think this crash is related to the one taking place here https://github.com/wordpress-mobile/WordPress-Android/issues/11260. I think the refactor [here](https://github.com/wordpress-mobile/WordPress-Android/issues/11271) should resolve this. 

* Sometimes when a `Scheduled` post is saved, it doesn't show the `Cancel` action or the title doesn't get updated. I don't think that's related to this PR but it might need investigation.

* I added a `localId` and `remoteId` to the `PageItem` because there are times when either is required. I renamed the `id` in `PageItem` to `localId` because it's often confusing when trying to determine which type of id should be used for a query 4caff1c. A further enhancement would be to simply change the types to `LocalId` and `RemoteId` so that that the compiler's type system enforces this across the featureset. I opted to leave this out of this PR. Should I create an issue for this or it can remain? since the name is already conveying the intent without changing the type. 

## Testing
1. Put the device in airplane mode.
2. Create a new page. Either a draft, published or scheduled page. Repeat all the steps below for each page type.
3. Once the page has been created, publish it or save it. 
4. Click on the more button and you should see a “Cancel” button. 
5. Once you press cancel you should see a snackbar message saying that the changes won’t be published. 
6. Go back online and ensure that the page isn’t uploaded. 
7. Also, check other pages that haven’t been changed to ensure the “Cancel” action doesn’t show for them. 

Before | After 
--------|-------
   <img src="https://user-images.githubusercontent.com/1509205/74075970-52932400-49e3-11ea-8931-cb6d55d8920c.png" width="320">     |    <img src="https://user-images.githubusercontent.com/1509205/74075972-545ce780-49e3-11ea-8d21-1290a25e6c7b.png" width="320">



## Reviewing 
Only 1 reviewer is needed but anyone can review. 

## Submitter Checklist
